### PR TITLE
Timer

### LINKED ISSUE #20: Save race times to filesystem
That way next time you load up the game you have your previous time records saved.

https://love2d.org/wiki/love.filesystem

Depends on: ~~#19~~, #32

### PR DESCRIPTION
**Pull request by SuperShadowPlay**
Closes #20 

- Adds a timer system to the game. When a map loads and is being played, a timer ticks away in the top-left corner. When the last checkpoint is cleared and the level ends, the time is saved to the save file.
These times are then displayed on the menu.
    - To save on screen space, the current loaded map is now indicated by turning the map selector red when the current map is selected.

- Adds a HUD file that currently just holds a few lines pertaining to the drawing of the in-game stopwatch. I am assuming we will probably want to add a few more HUD elements later on in development.
- Fixed bug where engine noise still plays on menu after the level is completed.

image

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -45,36 +46,32 @@ love.keypressed = function(pressed_key)
...
+    end
 
     Camera.unset()
 end
 
 love.update = function(dt)
     if State.paused == false then
         world:update(dt)
-        love.seconds = love.seconds + dt
+        State.seconds = State.seconds + dt
```
Comment (wyatt-herkamp) [inline main.lua:67]: Yall got rid of my bad idea

**Review by wyatt-herkamp — APPROVED** [review]
Looks Good
lovec_2022-02-15_22-37-29

File: menu/menu.lua [hunk 2]
```diff
@@ -15,21 +15,71 @@ menu.up = function()
...
+    -- Display three best times on record
+    for i = 0, 2 do
+        -- If map was completed and time is one of the best, display it in gold.
+        if menu.threeBest[i + 1] == convertedTime then
+            isNewBest = true
+            love.graphics.setColor(255, 215, 0, 1)
+        end
+        -- Print out times
+        love.graphics.print('#' .. i + 1 .. ': ' .. menu.threeBest[i + 1],
+                            corner[1] + 448, corner[2] + 430 + (i * 45), 0, 1.5, 1.5)
```
Comment (Jon-Cavaliere) [inline menu/menu.lua:53]: Love GUI formatting math. Very much more elegant than mine

**Review by Jon-Cavaliere — APPROVED** [review]
Looks great to me!

File: main.lua (hunk 1 above)
Comment (SuperShadowPlay) [reply main.lua:67]: Hey it's not a bad idea if all I did was change the name :)

File: menu/menu.lua (hunk 2 above)
Comment (SuperShadowPlay) [reply menu/menu.lua:53]: Yes I was particularly happy with that one. I live for little tricks like that.

**Pull request merged**